pyanaconda/modules/network/installation.py

```diff
@@ -33,10 +33,77 @@
 gi.require_version("NM", "1.0")
 from gi.repository import NM
 
+
+class HostnameConfigurationTask(Task):
+    """Hostname configuration task."""
+
+    HOSTNAME_CONF_FILE_PATH = "/etc/hostname"
+
+    def __init__(self, sysroot, hostname, overwrite):
+        """Create a new task.
+
+        :param sysroot: a path to the root of installed system
+        :type sysroot: str
+        :param hostname: static hostname
+        :type hostname: str
+        :param overwrite: overwrite config files if they already exist
+        :type overwrite: bool
+        """
+        super().__init__()
+        self._sysroot = sysroot
+        self._hostname = hostname
+        self._overwrite = overwrite
+
+    @property
+    def name(self):
+        return "Configure hostname"
+
+    def run(self):
+        self._write_hostname(self._sysroot, self._hostname, self._overwrite)
+
+    def _write_hostname(self, root, hostname, overwrite):
+        """Write static hostname to the target system configuration file.
+
+        :param root: path to the root of the target system
+        :type root: str
+        :param hostname: static hostname
+        :type hostname: str
+        :param overwrite: overwrite existing configuration file
+        :type overwrite: bool
+        """
+        return _write_config_file(root, self.HOSTNAME_CONF_FILE_PATH,
+                                  "{}\n".format(hostname),
+                                  "Cannot write hostname configuration file",
+                                  overwrite)
+
```

Comment:
I think you can inline this method to the run method.

Reply:
Done